contracts/validator-manager/ValidatorMessages.sol

```diff
@@ -162,11 +165,15 @@ library ValidatorMessages {
         // was tested against pre-allocating the array and doing manual byte by byte packing and
         // it was found to be more gas efficient.
         for (uint256 i = 0; i < subnetConversionData.initialValidators.length; i++) {
+            if (subnetConversionData.initialValidators[i].blsPublicKey.length != 48) {
```

Comment:
I know that we are doing this in other packing methods as well but I'm not convinced that we want to check this here.

With current usage this is not going to catch any user errors since the hash that we will verify against the hash of packed bytes is signed by the p-chain. 

I doubt that this would change the efficiency of the compiled bytecode but I would prefer to be careful with keeping the logic here as minimal as possible to make sure that the gas efficiency comment above this stays true. If we do want to check it still I would slightly prefer to do it in the `ValidatorManager.sol` where we loop through `validators` already.

Feel free to drop if I'm being paranoid 😅

Reply:
Agreed. We removed input validation on the `SubnetConversionData` parameter to `initializeValidatorSet` for the same reason. I've made this change here.